### PR TITLE
notice: prevent mutation of params

### PR DESCRIPTION
**Pull request by kyrylo**
Replaces #149 (Don't modify original params)

### CODE REVIEW

File: lib/airbrake-ruby/notice.rb [hunk 1]
```diff
@@ -240,6 +242,15 @@ def extract_custom_attributes(exception)
         )
       end
     end
+
+    def deep_dup(params, level = 1)
+      level = 0 if level > 3
```
Comment (vmihailenco) [inline lib/airbrake-ruby/notice.rb]: why this line? can't you instead `return params.dup if level > 3`

Comment (kyrylo) [inline lib/airbrake-ruby/notice.rb]: That'll work.

I spotted more serious issues with the approach (test failures).

**Comment by vmihailenco** [comment]
This always generates some garbage to dup params even if filters don't modify params. Instead you could try to carefully write filters so params are copied only if modification is needed and require users to do the same.

**Comment by kyrylo** [comment]
Closing because we discovered this is probably not the best approach to resolve the issue.

**Pull request closed without merging**